### DISCUSSION TITLE
Best practices for updating all translations when changing language

### DISCUSSION THREAD

**Discussion by kimiayahyazadeh** [amannn/next-intl#1097] [Q&A]
What is the recommended approach for updating all translations when the user changes the language via input in a Next.js application using next-intl? 
Any advice or examples on the best practices for this use case would be greatly appreciated.

**Comment by amannn** [comment]
It depends on if you use i18n routing or not. Does this help? [How can I change the locale?](https://next-intl-docs.vercel.app/docs/usage/configuration#locale-change)

If you don't use i18n routing, there's a strategy discussed in https://github.com/amannn/next-intl/discussions/1081 (see also: https://github.com/amannn/next-intl/discussions/1096)
